service/ui.go

```diff
@@ -388,8 +387,7 @@ func (uis *UIServer) GetServiceApp() *gimlet.APIApp {
 	app.AddRoute("/task_queue/{distro}/{task_id}").Wrap(needsLogin, needsContext).Handler(uis.legacyTaskQueue).Get()
 
 	// Patch pages
-	app.AddRoute("/patch/{patch_id}").Wrap(needsLogin, needsContext, viewTasks).Handler(uis.patchPage).Get()
-	app.AddRoute("/patch/{patch_id}").Wrap(needsLogin, needsContext, submitPatches).Handler(uis.schedulePatchUI).Post()
+	app.AddRoute("/patch/{patch_id}").Wrap(needsLogin).Handler(uis.legacyPatchPage).Get()
```

Comment:
Why does this remove needsContext and viewTasks?

Reply:
FWIW though since Spruce does its own permission checking, this is not strictly crucial anymore